code/datums/elements/weapon_description.dm

```diff
@@ -0,0 +1,185 @@
+
+	/// Doesn't show the base notes for items that have the override notes variable set to true
+	if(!source.override_notes)
+		/// Make sure not to divide by 0 on accident
+		if(source.force > 0)
+			readout += "Our extensive research has shown that it takes a mere <span class='warning'>[round((100 / source.force), 0.1)]</span> hits to down [victims[rand(1, victims.len)]] with no armor."
+		else
+			readout += "Our extensive research found that you couldn't beat yourself to death with this if you tried."
+
+		if(source.throwforce > 0)
+			readout += "If you decide to throw this object instead, one will take <span class='warning'>[round((100 / source.throwforce), 0.1)]</span> hits before collapsing."
+		else
+			readout += "If you decide to throw this object instead, then you will have trouble damaging anything."
+		if(source.armour_penetration > 0 || source.block_chance > 0)
+			readout += "This item has proven itself <span class='warning'>[weapon_tag_convert(source.armour_penetration)]</span> of piercing armor and <span class='warning'>[weapon_tag_convert(source.block_chance)]</span> of blocking attacks."
+	/// Custom manual notes
+	if(source.offensive_notes)
+		readout += source.offensive_notes
+
+	/// Spamming 'if's to see if we need to get additional weapon notes
+	if(istype(source, /obj/item/gun/ballistic))
+		readout += add_notes_ballistic(source)
+	else if(istype(source, /obj/item/gun/energy))
+		readout += add_notes_energy(source)
+	else if(istype(source, /obj/item/ammo_casing/))
+		readout += add_notes_ammo(source)
+
+	/// Finally bringing the fields together
+	return readout.Join("\n")
+
+/**
+ *
+ * Converts percentile based stats to an adjective appropriate for the
+ * examined warning label
+ *
+ * Arguments:
+ *  * tag_val: The value of the item to be added to the tag
+ */
+/datum/element/weapon_description/proc/weapon_tag_convert(var/tag_val)
+	switch(tag_val)
+		if(0)
+			return "INCAPABLE"
+		if(1 to 25)
+			return "BARELY CAPABLE"
+		if(26 to 50)
+			return "CAPABLE"
+		if(51 to 75)
+			return "VERY CAPABLE"
+		if(76 to INFINITY)
+			return "EXTREMELY CAPABLE"
+		else
+			return "STRANGELY CAPABLE"
+
+
+/**
+ *
+ * Outputs type-specific weapon stats for ballistic weaponry
+ *
+ * Arguments:
+ * 	* source - The object being evaluated
+ */
+/datum/element/weapon_description/proc/add_notes_ballistic(obj/item/gun/ballistic/source)
+	var/list/readout = list("")
+	if(source.magazine) /// Make sure you have a magazine, thats where the warning is!
+		readout += "Be especially careful around this device, as it is loaded with <span class='warning'>[source.magazine.caliber]</span> rounds, which you can inspect for more information."
+	else
+		readout += "The warning attached to the magazine is missing..."
+	return readout.Join("\n")
```

Comment:
When you move these over, just make these return the string directly. There's no point in making a 1 list string then joining it.